assets/app/scripts/directives/overviewDeployment.js

```diff
@@ -27,27 +28,39 @@ angular.module('openshiftConsole')
           $scope.desiredReplicas = null;
         });
 
-        // Debounce scaling so multiple clicks within 500 milliseconds only
-        // result in one request.
+        $scope.canScale = function() {
+          if (!$scope.deploymentConfigId || $scope.deploymentConfigMissing) {
+            // Scale as replication controller.
+            return true;
+          }
+
+          if (!$scope.deploymentConfig) {
+            // Wait for the deployment config.
+            return false;
+          }
+
+          // Only allow scaling of the latest deployment.
+          return deploymentIsLatestFilter($scope.rc, $scope.deploymentConfig);
```

Comment:
Not necessarily a new issue w/ this PR, but now that I'm looking at it, I think `deploymentIsLatest` is not what we want here. The "latest" deployment may not be the same as the "active" deployment. If you have a series of successful deployments and finally make a new deployment (version N) which fails, N is the latest but N-1 becomes active: version N is scaled down, version N-1 is scaled back up.

It might make sense to start making that distinction here, otherwise the user won't be able to adjust the scale of the active deployment (if they scale the "latest" which failed, the deployment controller will scale it back down again, as it does make the latest/active distinction.)

Make sense?